client/state/selectors/get-media.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns media for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {?Array}         Media
+ */
+export default function getMedia( state, siteId, query ) {
+	const queries = state.media.queries[ siteId ];
+
+	if ( ! queries ) {
+		return null;
+	}
+
+	return queries.getItems( query );
```

Comment:
@dmsnell Out of curiosity, how do you feel about Immutable.js objects in state? It seems quite similar in purpose to our Query Manager instances.

There was a balance in trying to implement Query Manager as a useful standalone module and incorporating it into state. The compromise I landed on was providing good serialization methods in the classes to still enable us to persist.

At its core, the class operates on two objects of data: `items` (keyed by the instance `options.itemKey`) and `queries` (serialized query objects to an array of keys). With your concerns in mind, these are perhaps the minimum "raw" state that a pure object store could be responsible for tracking, while the rest of its behaviors (getting, receiving) could accept these as arguments.

That said, I have a hunch this might come at the sacrifice of general usability of the class. I'm open to exploring and suggestions, though.